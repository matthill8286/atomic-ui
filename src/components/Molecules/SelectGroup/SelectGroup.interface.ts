import { MarginProperty } from 'csstype'
import { Corners, Shape, Surface } from '@/components/Atoms/Card/Card.interface'
import {
  Elevation,
  Padding,
  Position,
  ThemeColors,
  VerticalPadding,
  VerticalPaddingMap,
} from '@/types/theme'

export type SelectionDirection = 'row' | 'column'
export type SelectionState = 'default' | 'unselected' | 'selected' | 'disabled'
export type SelectionHandlerAction = 'add' | 'remove'
export type SelectionPosition = 'first' | 'last' | 'middle'

export type SelectGroupId = string
export type SelectedOption = SelectGroupId | null

export interface SelectItem {
  content: React.ReactNode
  disabled?: boolean
  id: SelectGroupId
}

export interface SelectGroupItemContainerProps {
  disabled: boolean
  surface: Surface
  textColor?: ThemeColors
  margin?: MarginProperty<string>
  showDivider?: boolean
  onClick: (event: React.MouseEvent<HTMLDivElement>) => void
}

export interface SelectGroupItemProps extends SelectItem {
  disabled?: boolean
  margin?: MarginProperty<string>
  elevation?: Elevation
  elevationHover?: Elevation
  noBorder: Position | Position[]
  onClick: (id: SelectGroupId, action: SelectionHandlerAction, event: React.MouseEvent) => void
  padding?: VerticalPaddingMap | VerticalPadding | Padding
  shape: Shape | Corners<Shape>
  surface: Surface
  textColor?: ThemeColors
  showDivider?: boolean
  dividerWidth?: string | number
}

export interface SelectGroupProps {
  direction?: SelectionDirection
  elevation?: Elevation
  elevationHover?: Elevation
  group: SelectItem[]
  onChange: (selected: SelectGroupId, event?: React.MouseEvent) => void
  padding?: VerticalPaddingMap | VerticalPadding | Padding
  selected: SelectedOption
  selectedElevation?: Elevation
  textColor?: ThemeColors
  showDivider?: boolean
  dividerWidth?: string | number
}

export interface SelectMultiGroupProps {
  direction?: SelectionDirection
  elevation?: Elevation
  elevationHover?: Elevation
  group: SelectItem[]
  onChange: (selected: SelectedOption[], event: React.MouseEvent) => void
  selected: SelectedOption[]
}
