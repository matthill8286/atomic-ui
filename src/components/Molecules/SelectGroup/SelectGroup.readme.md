## SelectGroup atom

Use selection groups like checkboxes or radio buttons when you have to present lots of information to the user.

### State

Default: In the default state the border is grey with the MediaMarkt black as the font-color.
Unselected: When one tile is selectd the other ones turn into the unselected state. The text color changes to a slightly lighter grey to diferentiate between selected and unselected.
Selected: When a tile gets selectd the state transitions to a red outline card with a slight shadow. The text-color stays in the MediaMarkt black.

#### #### Changes

Added the `elevation` and `elevationHover` props.
