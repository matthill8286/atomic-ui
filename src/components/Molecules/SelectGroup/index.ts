export * from './SelectGroup'
export * from './SelectMultiGroup'
