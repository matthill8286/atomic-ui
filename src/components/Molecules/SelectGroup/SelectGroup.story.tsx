import { select } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { Elevation, HorizontalPadding, Padding, VerticalPadding } from '@/types/theme'
import { SelectGroup } from './SelectGroup'
import { SelectGroupProps, SelectionDirection, SelectItem } from './SelectGroup.interface'
import Readme from './SelectGroup.readme.md'

const customPaddings: Padding = {
  bottom: 'xxs',
  left: 'sm',
  right: 'md',
  top: 'xl',
}

const paddings: VerticalPadding[] = ['xxs', 'xs', 'sm', 'md', 'lg', 'xl']

const directions: SelectionDirection[] = ['column', 'row']

const elevations: Elevation[] = [0, 1, 2, 3, 4]

const dividerWidths: string[] = [
  '10%',
  '20%',
  '30%',
  '40%',
  '50%',
  '60%',
  '70%',
  '80%',
  '90%',
  '100%',
]

const stories = storiesOf('Design System/Molecules/SelectGroup', module)

export const group: SelectItem[] = [
  { id: '1', content: <div>Option 1</div> },
  { id: '2', content: <div>Option 2</div> },
  { id: '3', content: <div>Option 3</div> },
  { id: '4', content: <div>Option 4</div> },
  { id: '5', content: <div>Option 5</div> },
]

stories.add(
  'Default',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 0),
      elevationHover: select('Elevation on Hover', elevations, 0),
      group,
      onChange: () => null,
      padding: select('Padding', paddings, 'xxs'),
      selected: null,
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'Selected',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 0),
      elevationHover: select('Elevation on Hover', elevations, 0),
      group,
      onChange: () => null,
      selected: '1',
      selectedElevation: select('Selected Elevation', elevations, 0),
      textColor: 'grey4',
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'Disabled',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 0),
      elevationHover: select('Elevation on Hover', elevations, 0),
      group: [...group, { id: '1', content: 'Option', disabled: true }],
      onChange: () => null,
      selected: '1',
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'One option',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 0),
      elevationHover: select('Elevation on Hover', elevations, 0),
      group: [{ id: 'vorkasse', content: 'Vorkasse' }],
      onChange: () => null,
      selected: null,
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'With elevations',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 1),
      elevationHover: select('Elevation on Hover', elevations, 2),
      group,
      onChange: () => null,
      selected: null,
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'Selected with elevations',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 1),
      elevationHover: select('Elevation on Hover', elevations, 2),
      group,
      onChange: () => null,
      selected: '1',
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'Selected with padding',
  () => {
    const horizontalPaddings: HorizontalPadding[] = ['xs', 'sm', 'md', 'lg', 'xl']
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 1),
      elevationHover: select('Elevation on Hover', elevations, 2),
      group,
      onChange: () => null,
      padding: {
        top: select('padding|paddingTop', paddings, customPaddings.top),
        bottom: select('padding|paddingBottom', paddings, customPaddings.bottom),
        left: select('padding|paddingLeft', horizontalPaddings, customPaddings.left),
        right: select('padding|paddingRight', horizontalPaddings, customPaddings.right),
      },
      selected: '1',
    }
    return <SelectGroup {...knobs} />
  },
  { info: Readme }
)

stories.add(
  'With Border width',
  () => {
    const knobs: SelectGroupProps = {
      direction: select('Direction', directions, 'column'),
      elevation: select('Elevation', elevations, 1),
      elevationHover: select('Elevation on Hover', elevations, 2),
      group,
      onChange: () => null,
      selected: null,
      dividerWidth: select('Divider Widths', dividerWidths, '70%'),
    }
    return <SelectGroup {...knobs} showDivider />
  },
  { info: Readme }
)
