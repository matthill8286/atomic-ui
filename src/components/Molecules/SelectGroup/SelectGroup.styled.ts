import { StyledDivider } from '@/components/Atoms/Card/Card.styled'
import { css, styled } from '@/styles/styled'
import { SelectGroupItemContainerProps } from './SelectGroup.interface'

export const SelectGroupItemContainer = styled.div<SelectGroupItemContainerProps>`
  cursor: pointer;
  position: relative;
  z-index: 1;

  &:hover {
    z-index: 2;
  }

  ${({ margin }) => margin && `margin: ${margin}`};

  ${({ surface }) =>
    surface === 'selected' &&
    css`
      cursor: default;
      z-index: 3;
    `}

  ${({ showDivider }) =>
    showDivider &&
    css`
      &:last-child {
        ${StyledDivider} {
          display: none;
        }
      }
    `}

  ${({ disabled }) =>
    disabled &&
    css`
      cursor: not-allowed;
      opacity: 0.7;
    `}
`
