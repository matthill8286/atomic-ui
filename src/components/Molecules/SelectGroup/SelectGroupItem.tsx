import * as React from 'react'
import { Card } from '@/components/Atoms/Card'
import { Padding } from '@/types/theme'
import { SelectGroupItemProps, SelectionHandlerAction } from './SelectGroup.interface'
import { SelectGroupItemContainer } from './SelectGroup.styled'

const cardPadding: Padding = {
  top: 'lg',
  bottom: 'lg',
  left: 'md',
  right: 'md',
}

export const SelectGroupItem: React.FC<SelectGroupItemProps> = ({
  content,
  disabled = false,
  elevation = 0,
  elevationHover = 0,
  showDivider = false,
  id,
  margin,
  noBorder,
  onClick,
  padding,
  shape,
  surface,
  dividerWidth,
  textColor,
  ...otherProps
}) => {
  const isSelected = surface === 'selected'
  const handleOnClick = (e: React.MouseEvent<HTMLDivElement>) => {
    e.stopPropagation()
    if (disabled) {
      return
    }
    const action: SelectionHandlerAction = isSelected ? 'remove' : 'add'
    onClick(id, action, e)
  }
  return (
    <SelectGroupItemContainer
      disabled={disabled}
      surface={surface}
      margin={margin}
      showDivider={showDivider}
      onClick={handleOnClick}
      {...otherProps}>
      <Card
        padding={padding || cardPadding}
        elevation={elevation}
        elevationHover={elevationHover}
        noBorder={noBorder}
        showDivider={showDivider}
        dividerWidth={dividerWidth}
        key={id}
        shape={shape}
        surface={surface}
        textColor={textColor}>
        {content}
      </Card>
    </SelectGroupItemContainer>
  )
}
