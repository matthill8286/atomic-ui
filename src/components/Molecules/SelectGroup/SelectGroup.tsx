import React, { useEffect, useState } from 'react'
import { Corners, Shape } from '@/components/Atoms/Card/Card.interface'
import { FlexBox } from '@/components/Helper/FlexBox'
import { isAlternateTheme } from '@/utils/helper'
import {
  SelectedOption,
  SelectGroupId,
  SelectGroupProps,
  SelectionHandlerAction,
  SelectItem,
} from './SelectGroup.interface'
import { SelectGroupItem } from './SelectGroupItem'
import {
  defaultShape,
  getItemBorder,
  getItemIndex,
  getItemPosition,
  getItemShape,
  getItemState,
  mapStateToSurface,
} from './utils'

// TODO: update the SelectGroup to have a checkbox behaviour
export const SelectGroup: React.FC<SelectGroupProps> = ({
  direction = 'column',
  elevation = 0,
  elevationHover = 0,
  group,
  onChange,
  padding,
  selected,
  selectedElevation = 0,
  showDivider = isAlternateTheme(),
  dividerWidth,
  textColor,
  ...otherProps
}) => {
  const [selection, changeSelection] = useState<SelectedOption>(selected)

  useEffect(() => {
    if (selected !== selection) {
      changeSelection(selected)
    }
  }, [group, selected])

  const addItem = (id: SelectGroupId): void => {
    changeSelection(id)
  }

  const handleOnClick = (
    id: SelectGroupId,
    action: SelectionHandlerAction,
    event: React.MouseEvent
  ) => {
    if (action === 'add') {
      addItem(id)
    }
    // TODO: for the checkbox behavior a removeItem is required
    onChange(id, event)
  }

  const hasMoreThanOne = group.length > 1

  return (
    <FlexBox flexDirection={direction} {...otherProps}>
      {group.map(({ content, disabled = false, id }: SelectItem, index: number) => {
        const state = getItemState(id, selection, disabled)
        const hasSelectedElevation = state === 'selected' && selectedElevation
        const position = getItemPosition(index + 1, group.length)
        const selectedIndex = getItemIndex(selection, group)
        const noBorder = hasMoreThanOne
          ? getItemBorder(direction, position, index, selectedIndex)
          : []
        const shape: Corners<Shape> = hasMoreThanOne
          ? getItemShape(direction, position)
          : defaultShape
        const surface = mapStateToSurface(state)
        const itemElevation = hasSelectedElevation ? selectedElevation : elevation
        const itemElevationHover = hasSelectedElevation ? undefined : elevationHover

        return (
          <SelectGroupItem
            elevation={itemElevation}
            elevationHover={itemElevationHover}
            content={content}
            disabled={disabled}
            id={id}
            key={id}
            noBorder={noBorder}
            onClick={handleOnClick}
            padding={padding}
            shape={shape}
            surface={surface}
            textColor={textColor}
            showDivider={showDivider}
            dividerWidth={dividerWidth}
          />
        )
      })}
    </FlexBox>
  )
}
