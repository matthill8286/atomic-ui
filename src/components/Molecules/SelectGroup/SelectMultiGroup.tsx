import React, { FC, useEffect, useState } from 'react'
import { FlexBox } from '@/components/Helper/FlexBox'
import {
  SelectedOption,
  SelectGroupId,
  SelectionHandlerAction,
  SelectItem,
  SelectMultiGroupProps,
} from './SelectGroup.interface'
import { SelectGroupItem } from './SelectGroupItem'

export const SelectMultiGroup: FC<SelectMultiGroupProps> = props => {
  const {
    direction = 'row',
    elevation = 0,
    elevationHover = 0,
    group = [],
    selected = [],
    onChange,
    ...otherProps
  } = props

  const [selection, changeSelection] = useState<SelectedOption[]>(selected)

  useEffect(() => {
    if (!selected.find(sel => selection.includes(sel))) {
      changeSelection(selected)
    }
  }, [group, selected])

  const handleClick = (
    id: SelectGroupId,
    action: SelectionHandlerAction,
    event: React.MouseEvent
  ) => {
    let changed: SelectedOption[] = []
    if (action === 'add') {
      changed = [...selection, id]
    }
    if (action === 'remove') {
      changed = selection.filter(element => element !== id)
    }
    changeSelection(changed)
    onChange(changed, event)
  }

  const isOptionActive = (item: SelectItem) => {
    return selection.includes(item.id)
  }

  return (
    <FlexBox flexDirection={direction} {...otherProps}>
      {group.map((option: SelectItem) => (
        <SelectGroupItem
          id={option.id}
          key={option.id}
          elevation={elevation}
          elevationHover={elevationHover}
          margin={direction === 'row' ? '0 12px' : '12px 0'}
          content={option.content}
          disabled={option.disabled}
          noBorder={[]}
          onClick={handleClick}
          shape="rounded-small"
          surface={isOptionActive(option) ? 'selected' : 'white'}
        />
      ))}
    </FlexBox>
  )
}
