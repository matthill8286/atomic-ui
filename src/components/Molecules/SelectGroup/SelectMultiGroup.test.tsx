import * as React from 'react'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { SelectItem, SelectMultiGroupProps } from './SelectGroup.interface'
import { SelectMultiGroup } from './SelectMultiGroup'

export const group: SelectItem[] = [
  { id: '1', content: <div>Option 1</div> },
  { id: '2', content: <div>Option 2</div> },
  { id: '3', content: <div>Option 3</div> },
  { id: '4', content: <div>Option 4</div> },
  { id: '5', content: <div>Option 5</div> },
]

describe('SelectMultiGroup', () => {
  const props: SelectMultiGroupProps = {
    group,
    onChange: jest.fn(),
    selected: [],
  }

  it('renders correctly', () => {
    const tree = renderWithTheme(<SelectMultiGroup {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with direction row', () => {
    const tree = renderWithTheme(<SelectMultiGroup {...props} direction="row" />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with direction column', () => {
    const tree = renderWithTheme(<SelectMultiGroup {...props} direction="column" />)
    expect(tree).toMatchSnapshot()
  })

  it('mounts the group elements', () => {
    const component = mountWithTheme(<SelectMultiGroup {...props} />)
    expect(component.find('SelectGroupItem')).toHaveLength(5)
  })

  it('should render with elements selected', () => {
    const component = mountWithTheme(<SelectMultiGroup {...props} selected={['1', '2', '5']} />)
    expect(component).toMatchSnapshot()
  })
})
