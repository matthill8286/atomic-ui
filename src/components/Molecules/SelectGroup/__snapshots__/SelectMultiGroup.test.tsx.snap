// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMultiGroup renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  padding-top: 32px;
  padding-right: 24px;
  padding-bottom: 32px;
  padding-left: 24px;
  color: #000000;
  border: 1px solid #cfcbca;
  background-color: #fff;
  border-radius: 4px;
  height: 100%;
}

.c2 {
  position: relative;
  outline: none;
}

.c2 .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c1 {
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin: 0 12px;
}

.c1:hover {
  z-index: 2;
}

@media (min-width:512px) {
  .c2:hover .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0,
  .c2:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1b4w28x-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<div
  className="c0"
  display="flex"
>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 1
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 2
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 3
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 4
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 5
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectMultiGroup renders with direction column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  padding-top: 32px;
  padding-right: 24px;
  padding-bottom: 32px;
  padding-left: 24px;
  color: #000000;
  border: 1px solid #cfcbca;
  background-color: #fff;
  border-radius: 4px;
  height: 100%;
}

.c2 {
  position: relative;
  outline: none;
}

.c2 .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c1 {
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin: 12px 0;
}

.c1:hover {
  z-index: 2;
}

@media (min-width:512px) {
  .c2:hover .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0,
  .c2:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1b4w28x-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<div
  className="c0"
  display="flex"
>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 1
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 2
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 3
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 4
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 5
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectMultiGroup renders with direction row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  padding-top: 32px;
  padding-right: 24px;
  padding-bottom: 32px;
  padding-left: 24px;
  color: #000000;
  border: 1px solid #cfcbca;
  background-color: #fff;
  border-radius: 4px;
  height: 100%;
}

.c2 {
  position: relative;
  outline: none;
}

.c2 .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c1 {
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin: 0 12px;
}

.c1:hover {
  z-index: 2;
}

@media (min-width:512px) {
  .c2:hover .Cardstyled__StyledBadgeLineWrapper-sc-1b4w28x-0,
  .c2:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1b4w28x-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<div
  className="c0"
  display="flex"
>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 1
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 2
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 3
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 4
        </div>
      </div>
    </div>
  </div>
  <div
    className="c1"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        display="block"
        elevation={0}
        shape="rounded-small"
      >
        <div>
          Option 5
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectMultiGroup should render with elements selected 1`] = `ReactWrapper {}`;
